shell/common/platform_view.cc

```diff
@@ -140,6 +140,33 @@ void PlatformView::HandlePlatformMessage(
     response->CompleteEmpty();
 }
 
+size_t PlatformView::RegisterTexture(std::shared_ptr<flow::Texture> texture) {
+  int texture_id;
+  fxl::AutoResetWaitableEvent latch;
+  blink::Threads::Gpu()->PostTask([this, &texture_id, &latch, texture]() {
+    texture_id = rasterizer_->GetTextureRegistry().RegisterTexture(texture);
+    latch.Signal();
+  });
+  latch.Wait();
+  return texture_id;
+}
+
+void PlatformView::UnregisterTexture(size_t texture_id) {
+  fxl::AutoResetWaitableEvent latch;
+  blink::Threads::Gpu()->PostTask([this, &latch, texture_id]() {
+    rasterizer_->GetTextureRegistry().UnregisterTexture(texture_id);
+    latch.Signal();
+  });
+  latch.Wait();
+}
+
+void PlatformView::MarkTextureFrameAvailable(size_t texture_id) {
+  blink::Threads::UI()->PostTask([engine = engine_->GetWeakPtr()] {
+    if (engine)
+      engine->ScheduleRedraw();
```

Comment:
I am pretty sure this engine call can be made from any thread. The animator will do the job of scheduling the awaits onto the correct thread.